src/main/java/br/andre/estudos/strings/StringProcessamento.java

```diff
@@ -33,25 +42,78 @@ public void exibeMenu() {
                     out.println("A quantidade de caracteres é " + quantidadeCaracteres(texto));
                     break;
                 case 2:
-                    out.println("Digite um texto:");
+                    out.print("Digite um texto:");
                     texto = sc.nextLine();
                     out.println("O texto invertido fica " + inverteTexto(texto));
                     break;
                 case 3:
-                    out.println("Digite um texto:");
+                    out.print("Digite um texto:");
                     texto = sc.nextLine();
-                    out.println("O texto em maiúsculo fica "+ textoMaiusculo(texto));
+                    out.println("O texto em maiúsculo fica " + textoMaiusculo(texto));
                     break;
                 case 4:
-                    out.println("Digite um texto:");
+                    out.print("Digite um texto:");
                     texto = sc.nextLine();
-                    out.println("O texto em minúsculo fica "+ textoMinusculo(texto));
+                    out.println("O texto em minúsculo fica " + textoMinusculo(texto));
                     break;
                 case 5:
-                    out.println("Digite um texto:");
+                    out.print("Digite um texto:");
                     texto = sc.nextLine();
                     out.println("O texto com a primeira letra maiúscula fica " + upperCaseFirst(texto));
                     break;
+                case 6:
+                    out.println("Digite um texto:");
+                    texto = sc.nextLine();
+                    String resultado = textoPalindromo(texto) ? "é palíndromo" : "não é palíndromo";
+                    out.printf("O texto " + resultado);
+                    break;
+                case 7:
+                    out.print("Digite um texto:");
+                    texto = sc.nextLine();
+
+                    if (quantidadePalavra(texto) > 1) {
+                        String s = "O texto tem " + quantidadePalavra(texto) + " palavra(s)";
+                        out.print(s);
+                    } else {
+                        String s = "O texto tem " + quantidadePalavra(texto) + " palavra";
+                        out.print(s);
+                    }
+                    break;
+                case 8:
+                    out.print("Digite um texto:");
+                    texto = sc.nextLine();
+                    out.println(textoSemEspaco(texto));
+                    break;
+                case 9:
+                    out.print("Digite um texto:");
+                    texto = sc.nextLine();
+                    out.print("Digite a posição do caractere:");
+                    posicaoTexto = sc.nextInt();
+                    out.println("O seu caractere é " + caractereDePosicao(texto, posicaoTexto));
```

Comment:
E se o usuário colocar uma palavra pequena e informar uma posição inexistente?

Reply:
@AndreSchuh Aqui não resolveu.
Você adicionou o `if`certo, mas tem que colocar dentro do método `caractereDePosicao` , senão vai continuar dando erro
![image](https://github.com/AndreSchuh/java-test/assets/29099986/6aece69a-67e3-4e3d-ad1e-280c15cead34)

Testa melhor esse ponto, vê se tu entende.

Reply:
Ve se assim fica melhor, não consegui adicionar um return de "opção inválida"

Reply:
Acho que resolvi porém não sei da onde saiu o 9

Reply:
Esse também tem o problema da letra maiúscula porém não consegui incluir com a expressão regular e muito menos com da outra maneira será que tu conseguiria me explicar isso aí? Por que 2 if junto com um for pra mim é loucura.